src/HyperfundFactory.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.28;
+
+import "./Hyperfund.sol"; // Import Hyperfund contract
+import "./Hyperstaker.sol"; // Import Hyperstaker contract
+
+contract HyperfundFactory {
+    Hyperfund[] public hyperfunds; // Array to store deployed Hyperfund contracts
+    Hyperstaker[] public hyperstakers; // Array to store deployed Hyperstaker contracts
```

Comment:
also, why the need to store these?

Reply:
Do we need to retrieve the list of all deployed contracts by factory at contract level? If not I can remove this as it does not serve any other purpose.

Reply:
i don't see why we need this on chain